qcodes/utils/metadata.py

```diff
@@ -114,15 +115,22 @@ def diff_param_values(left_snapshot: Snapshot,
         }
     )
 
-def diff_param_values_by_id(left_id : RunId, right_id : RunId):
+
+def diff_param_values_by_id(left_id: RunId, right_id: RunId) -> ParameterDiff:
     """
     Given the IDs of two datasets, returns the differences between
     parameter values in each of their snapshots.
     """
     # Local import to reduce load time and
     # avoid circular references.
     from qcodes.dataset.data_set import load_by_id
-    return diff_param_values(
-        load_by_id(left_id).snapshot,
-        load_by_id(right_id).snapshot
-    )
+
+    left_snapshot = load_by_id(left_id).snapshot
+    right_snapshot = load_by_id(right_id).snapshot
+
+    if left_snapshot is None or right_snapshot is None:
+        raise RuntimeError(f"Tried to compare {left_snapshot} "
+                           f"and {right_snapshot}"
+                           f"but on of them is empty")
```

Comment:
I am not sure how to handle the case of one or both of the snapshots being empty? For now I just made it raise

Reply:
it's good for now. Just one remark: in case one of them is not None, do we really want that full dictionary to be printer in the exception message?